### PR TITLE
Improve user config migration implementation

### LINKED ISSUE #81: Improve user config migration implementation
### Description

The current implementation for `migrateUserConfig` can be improved if the following requirements in the acceptance criteria are met.

### Acceptance Criteria

- The existing code implementation is refactored
- All possible entries from Lisk Core v3 config are migrated to Lisk Core v4 compatible config (`PartialApplicationConfig`)
  - TODO: Specify all possible mappings
- Every migrated property is logged so that the user is aware of the same
- Tests are up-to-date

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #81 

### How was it solved?

- [x] The existing code implementation is refactored
- [x] All possible entries from Lisk Core v3 config are migrated to Lisk Core v4 compatible config 
- [x] Every migrated property is logged 
- [x] Add implementation to get user confirmation before starting core
- [x] Fix/add tests

### How was it tested?
Local

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -311,20 +308,31 @@ class LiskMigrator extends Command {
...
-						finalConfigCorev4 = configCoreV4;
+						finalConfigCorev4 = configV4;
+					}
+					const isUserConfirmed = await cli.confirm(
+						`Starting application using the configuration: ${util.inspect(
+							finalConfigCorev4,
+							false,
+							3,
+						)} [yes/no]`,
+					);
```
Comment (sameersubudhi) [inline src/index.ts:323]:
```suggestion
					const isUserConfirmed = await cli.confirm(`
						Start Lisk Core with the following configuration? [yes/no] \n
						${util.inspect(finalConfigCorev4, false, 3)}
					`);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: src/utils/config.ts [hunk 2]
```diff
@@ -138,11 +241,9 @@ export const validateConfig = async (config: ApplicationConfig): Promise<boolean
 };
 
 export const writeConfig = async (config: ApplicationConfig, outputPath: string): Promise<void> => {
-	if (existsSync(outputPath)) {
-		unlinkSync(outputPath);
+	if (!existsSync(outputPath)) {
```
[the remark this answers is not in the document]
Comment (sameersubudhi) [reply src/utils/config.ts:244]: Clarified offline!

File: src/types.ts [hunk 3]
```diff
@@ -20,6 +20,11 @@ interface RPCConfig {
 }
 
 export interface NetworkConfig {
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply src/types.ts:22]: I checked and some are not exported from the SDK.

File: src/constants.ts [hunk 4]
```diff
@@ -74,11 +74,12 @@ export const DEFAULT_HOST = '127.0.0.1';
...
-export const KEEP_EVENTS_FOR_HEIGHTS = 300;
 export const EMPTY_BYTES = Buffer.alloc(0);
 export const SHA_256_HASH_LENGTH = 32;
 export const BINARY_ADDRESS_LENGTH = 20;
 export const TRANSACTION_ID_LENGTH = SHA_256_HASH_LENGTH;
 
 export const DEFAULT_DATA_DIR = 'data';
 export const EXTRACTED_SNAPSHOT_DIR = 'blockchain.db';
+
+export const DEFAULT_VERSION = '0.1.0';
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply src/constants.ts:85]: I confirmed with SDK and default version is `0.1.0`

**Review by sameersubudhi — APPROVED** [review]

**Review by vardan10 — APPROVED** [review]

**Pull request merged**